dask/base.py

```diff
@@ -416,14 +416,32 @@ def tokenize(*args, **kwargs):
     return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
 
 
+def dont_optimize(dsk, keys):
+    return dsk
+
+
+def optimization_function(obj):
```

Comment:
I might move this to a property of the `Base` class, but this also could be fine. There are a few places in the code where we explicitly call the `_optimize` method of a class (e.g.:https://github.com/dask/dask/blob/master/dask/delayed.py#L76). Having a quick method on the class may be slightly more concise than calling a function on the class.

Alternatively, an explicit decorator as @shoyer mentioned on our base classes might be nice, as the existing `_optimize` calls would still work, but would be overideable. Untested:

```python
@curry
def defer_to_override(name, default):
    def _(dsk, keys):
        if name in _globals:
            f = _globals[name] or dont_optimize
            return f(dsk, keys)
        return default(dsk, keys)
    return _

class Array(Base):
    @staticmethod
    @defer_to_override('array_optimize')
    def _optimize(dsk, keys):
        ...
```